Plugins/Public/base_plugin/Main.cpp

```diff
@@ -2757,3 +2788,33 @@ EXPORT PLUGIN_INFO* Get_PluginInfo()
 	p_PI->lstHooks.push_back(PLUGIN_HOOKINFO((FARPROC*)&Plugin_Communication_CallBack, PLUGIN_Plugin_Communication, 11));
 	return p_PI;
 }
+
+void ResetShieldStrength() {
+	for (map<uint, PlayerBase*>::iterator i = player_bases.begin(); i != player_bases.end(); ++i) {
+		i->second->ResetShieldStrength();
+	}
+}
+
+bool checkBaseVulnerabilityStatus() {
+
+	if (baseVulnerabilityWindows.empty()) {
+		return false;
+	}
+
+	time_t tNow = time(0);
+	struct tm *t = localtime(&tNow);
+	uint currHour = t->tm_hour;
+	for (list<BASE_VULNERABILITY_WINDOW>::iterator i = baseVulnerabilityWindows.begin(); i != baseVulnerabilityWindows.end(); ++i){
+		if((i->start < i->end 
```

Comment:
Let's leave a comment about what this very messy if statement trying to do. I think the main distinction this code is looking for is whether a vulnerability window is defined as e.g.:
* more usually e.g. 0100-0700 - in which cases bases are vulneable when the time is on/after 0100 AND before 0700
* slightly less obvious problem case e.g. 2100-0300, i.e. over a day boundary - in which case bases are vulnerable when the time is on/after 2100 OR before 0300.

Reply:
Added the comment